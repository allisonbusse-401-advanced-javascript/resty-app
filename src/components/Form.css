.Form {
  padding-top: 40px;

  input[type="radio" i] {
    display: none;
  }
  input[type="text"] {
    width: 900px;
    line-height: 20px;
    margin-bottom: 20px;
  }
  label {
    margin-right: 6px;
    margin: 5px 6px 5px 0;
    cursor: pointer;
    padding: 3px;
    color: black;
    font-family: Arial, Helvetica, sans-serif;
  }
  label:hover {
    background-color: lightgray;
  }
  input:checked + label {
    background-color: #2D5986;
    color: white;
  }
  textarea {
    margin-top: 20px;
    width: 400px;
    height: 100px;
  }
}
.radioButtons {
  display: flex;
  justify-content: space-between;
  padding: 15px 0;
  }
