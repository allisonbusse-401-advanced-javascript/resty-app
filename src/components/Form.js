import React from 'react';
import styles from './Form.css';
import PropTypes from 'prop-types';

const Form = ({ handleSubmit, handleChange, url, method, body }) => (
  <form className={styles.Form} onSubmit={handleSubmit}>
    <input type="text" name="url" placeholder="URL" value={url} onChange={handleChange}></input>
    <div className="radioButtons">
      <input type="radio" id="get" name="method" value="get" checked={method === 'get'} onChange={handleChange}></input>
      <label htmlFor="get">
        GET</label>

      <input type="radio" id="post" name="method" value="post" checked={method === 'post'} onChange={handleChange}></input>
      <label htmlFor="post">
        POST</label>

      <input type="radio" id="put" name="method" value="put" checked={method === 'put'} onChange={handleChange}></input>
      <label htmlFor="put">
        PUT</label>
      <input type="radio" name="method" id="patch" value="patch" checked={method === 'patch'} onChange={handleChange}></input>
      <label htmlFor="patch">
        PATCH</label>

      <input type="radio" id="delete" name="method" value="delete" checked={method === 'delete'} onChange={handleChange}></input>
      <label htmlFor="delete">
        DELETE</label>
    </div>
    <textarea name="body" placeholder="Raw JSON Body" value={body} onChange={handleChange}></textarea>
    <button>Go!</button>
  </form>

);

Form.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
};

export default Form;
