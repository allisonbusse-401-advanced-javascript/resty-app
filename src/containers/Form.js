import React, { Component } from 'react';
import ResultsDisplay from '../components/ResultsDisplay';

export default class Form extends Component {

  state = {
    url: '',
    method: 'get',
    body: '',
    results: {
      headers: '{}',
      response: '{}'
    },
    loading: false
  }

  handleSubmit() {

  }

  handleChange() {

  }

  render() {
    return (
      <>
        <form onSubmit={this.handleSubmit}>
          <input type="text" name="url" placeholder="URL" value={this.state.url} onChange={this.handleChange}></input>
          <div className="radioButtons">
            <label>
              GET
              <input type="radio" value="get" checked={this.state.method === 'get'} onChange={this.handleChange}></input>
            </label>
            <label>
              POST
              <input type="radio" value="post" checked={this.state.method === 'post'} onChange={this.handleChange}></input>
            </label>
            <label>
              PUT
              <input type="radio" value="put" checked={this.state.method === 'put'} onChange={this.handleChange}></input>
            </label>
            <label>
              PATCH
              <input type="radio" value="patch" checked={this.state.method === 'patch'} onChange={this.handleChange}></input>
            </label>
            <label>
              DELETE
              <input type="radio" value="delete" checked={this.state.method === 'delete'} onChange={this.handleChange}></input>
            </label>
          </div>
          <textarea name="body" placeholder="Raw JSON Body" value={this.state.body} onChange={this.handleChange}></textarea>
          <button>Go!</button>
        </form>

        <ResultsDisplay
          headers={this.state.results.headers}
          response={this.state.results.response}
        />


      </>
    );
  }

  static propTypes = {

  }
}
