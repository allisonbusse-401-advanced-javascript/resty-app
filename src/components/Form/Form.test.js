import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders Form', () => {   
    const props = {
      handleSubmit: () => {},
      handleChange: () => {},
      url: 'jkl',
      method: 'jkl',
      body: 'uio'
    };

    const wrapper = shallow(<Form {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
