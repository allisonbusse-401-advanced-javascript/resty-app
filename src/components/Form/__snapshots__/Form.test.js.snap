// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <input
    name="url"
    onChange={[Function]}
    placeholder="URL"
    type="text"
    value="jkl"
  />
  <div
    className="radioButtons"
  >
    <input
      checked={false}
      id="get"
      name="method"
      onChange={[Function]}
      type="radio"
      value="get"
    />
    <label
      htmlFor="get"
    >
      GET
    </label>
    <input
      checked={false}
      id="post"
      name="method"
      onChange={[Function]}
      type="radio"
      value="post"
    />
    <label
      htmlFor="post"
    >
      POST
    </label>
    <input
      checked={false}
      id="put"
      name="method"
      onChange={[Function]}
      type="radio"
      value="put"
    />
    <label
      htmlFor="put"
    >
      PUT
    </label>
    <input
      checked={false}
      id="patch"
      name="method"
      onChange={[Function]}
      type="radio"
      value="patch"
    />
    <label
      htmlFor="patch"
    >
      PATCH
    </label>
    <input
      checked={false}
      id="delete"
      name="method"
      onChange={[Function]}
      type="radio"
      value="delete"
    />
    <label
      htmlFor="delete"
    >
      DELETE
    </label>
    <button>
      Go!
    </button>
  </div>
  <textarea
    name="body"
    onChange={[Function]}
    placeholder="Raw JSON Body"
    value="uio"
  />
</form>
`;
